directorywatcher/directorywatcher.go

```diff
@@ -0,0 +1,288 @@
+package directorywatcher
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	"gopkg.in/fsnotify.v1"
+
+	"github.com/reddit/baseplate.go/filewatcher"
+	"github.com/reddit/baseplate.go/internal/limitopen"
+	"github.com/reddit/baseplate.go/log"
+)
+
+// DirectoryWatcher loads and parses data from a file and watches for changes to that
+// file in order to refresh it's stored data.
+type DirectoryWatcher interface {
```

Comment:
"Provide concrete types, accept interfaces."

In this case, you are providing `*Result` and thus don't need to export the interface that it implements.  I don't think you use this interface at all, so it can probably be removed.  Callers can define their own interface if they need to abstract over a subset of its behavior.